src/test/java/uk/gov/pay/directdebit/payments/services/GoCardlessServiceTest.java

```diff
@@ -139,55 +124,65 @@ public void findMandateForEvent_shouldThrowIfNoMandateIsFoundForEvent() {
-        when(mockedGoCardlessClientWrapper.createCustomerBankAccount(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessCustomer, payer.getName(), SORT_CODE, ACCOUNT_NUMBER))
+        when(mockedGoCardlessClientWrapper.createCustomerBankAccount(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessCustomer, payer.getName(), payer.getSortCode(), payer.getAccountNumber()))
                 .thenThrow(new RuntimeException("oops"));
         thrown.expect(CreateCustomerBankAccountFailedException.class);
         thrown.expectMessage(format("Failed to create customer bank account in gocardless, payment request id: %s, payer id: %s", PAYMENT_REQUEST_EXTERNAL_ID, payer.getExternalId()));
         thrown.reportMissingExceptionWithMessage("CreateCustomerBankAccountFailedException expected");
-        service.createPayer(PAYMENT_REQUEST_EXTERNAL_ID, gatewayAccount, CREATE_PAYER_REQUEST);
+        service.confirm(PAYMENT_REQUEST_EXTERNAL_ID, gatewayAccount);
     }
 
     @Test
-    public void confirm_shouldStoreAGoCardlessMandateAndPaymentWhenReceivingConfirmPaymentRequest() {
+    public void confirm_shouldStoreAGoCardlessCustomerBankAccountMandateAndPaymentWhenReceivingConfirmPaymentRequest() {
         when(mockedGoCardlessCustomerDao.findByPayerId(payer.getId())).thenReturn(Optional.of(goCardlessCustomer));
-        when(mockedPaymentConfirmService.confirm(gatewayAccount.getExternalId(), PAYMENT_REQUEST_EXTERNAL_ID)).thenReturn(confirmationDetails);
+        when(mockedGoCardlessClientWrapper.createCustomer(PAYMENT_REQUEST_EXTERNAL_ID, payer)).thenReturn(goCardlessCustomer);
+        when(mockedGoCardlessClientWrapper.createCustomerBankAccount(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessCustomer, payer.getName(), payer.getSortCode(), payer.getAccountNumber())).thenReturn(goCardlessCustomer);
         when(mockedGoCardlessClientWrapper.createMandate(PAYMENT_REQUEST_EXTERNAL_ID, mandate, goCardlessCustomer)).thenReturn(goCardlessMandate);
         when(mockedGoCardlessClientWrapper.createPayment(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessMandate, transaction)).thenReturn(goCardlessPayment);
         when(mockedPayerService.getPayerFor(transaction)).thenReturn(payer);
         service.confirm(PAYMENT_REQUEST_EXTERNAL_ID, gatewayAccount);
+        verify(mockedGoCardlessCustomerDao).insert(goCardlessCustomer);
         verify(mockedGoCardlessMandateDao).insert(goCardlessMandate);
         verify(mockedGoCardlessPaymentDao).insert(goCardlessPayment);
         verify(mockedGoCardlessClientWrapper).createMandate(PAYMENT_REQUEST_EXTERNAL_ID, mandate, goCardlessCustomer);
         verify(mockedGoCardlessClientWrapper).createPayment(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessMandate, transaction);
+        verify(mockedGoCardlessClientWrapper).createCustomer(PAYMENT_REQUEST_EXTERNAL_ID, payer);
+        verify(mockedGoCardlessClientWrapper).createCustomerBankAccount(PAYMENT_REQUEST_EXTERNAL_ID, goCardlessCustomer, payer.getName(), payer.getSortCode(), payer.getAccountNumber());
```

Comment:
Since the customer has to be created before the customer bank account, I spot an opportunity for in-order verification here!

Reply:
Good catch!